certauth.go

```diff
@@ -197,22 +234,30 @@ func (a *Auth) CheckAuthorization(
 		err    error
 	)
 
-	for _, ck := range a.opt.AuthorizationCheckers {
-		if ps == nil { // not using httprouter
-			params, err = ck.CheckAuthorization(ou, cn)
-		} else { // using httprouter
-			params, err = ck.CheckAuthorizationWithParams(ou, cn, ps)
-		}
-		if err != nil {
-			return nil, err
-		}
-
-		// Collect the context params from each AuthorizationChecker into one map
-		if params != nil {
-			for k, v := range params {
-				ctxParams[k] = v
+	if len(a.opt.AuthorizationCheckers) > 0 {
+		a.checkers = append(a.checkers, a.opt.AuthorizationCheckers)
+	}
```

Comment:
This should be in the `New` function instead, I think, otherwise the `a.checkers` is gonna grow indefinitely each time `CheckAuthorization` is called.

Alternatively, you could clear `a.opt.AuthorizationCheckers` after appending it to `a.checkers` but I think it makes more sense in the constructor since that's definitely only run once.

Reply:
Yeah, we can't just do this in `New` because that would break compatibility with the deprecated `NewAuth`, so I think introducing a new variable scoped to just this function is the solution.